### ISSUE TITLE
Revisit and Refine Next Build Template Migration Process

### ISSUE THREAD

**Issue by gracekretschmer-metrostar** [department-of-veterans-affairs/va.gov-cms#20129]
## User Story or Problem Statement
Now that we have gone through the events and event listing launch, we want to reflect and refine the process for launching future content types and templates into prod Next Build. 

## Steps for Implementation
- Review how the events launch went and determine what we do and do not want to move forward.
- Collaborate with CMS PO on these findings.
- Synthesizing findings into an updated launch plan.
- Share updated launch plan with impacted teams.

## Acceptance Criteria
- [x] An established agreement and understanding of ownership and expectations between CMS and Sitewide.
- [x] An updated migration process is added in slack, github, and/or confluence. 
- [x] An understanding across teams of which Github projects and Slack channels will be used moving forward for the template migration work. 
- [x] The teams impacted are aware of any process changes and the new launch plan.

**Comment by gracekretschmer-metrostar** [comment]
@timcosgrove I am leaning towards pausing this work for sprint 26 until the new team comes onboard.

**Comment by gracekretschmer-metrostar** [comment]
Pivot this ticket to be including guidance on PR review process and expectations.

**Comment by laflannery** [comment]
**Update 2/5:**


- An established agreement and understanding of ownership and expectations between CMS and Sitewide.
  - CMS will own the “plumbing” of next-build
  - **IN PROGRESS:** RACI for the migration itself has been added to [Slack canvas](https://dsva.slack.com/canvas/CT4GZBM8F) for review and approval by CMS and Sitewide stakeholders
    - Currently waiting on reviews/approval by Erika Washburn, Michael Skinner and Tim Cosgrove for this to be finalized
  - CMS team will own clean up from Events next-build launch (QA tickets that were not completed, content-build removal)
  - Sitewide will own all templates post-migration

- An updated migration process is added in slack, github, and/or confluence.
  - IN PROGRESS: Current Draft process:
    - All migration tickets are in the Pre-migration column in the [Next-build GitHub Project board](https://github.com/orgs/department-of-veterans-affairs/projects/1464/views/1)
      - If a ticket is in this column, it has been refined and is ready to be picked up
    - When a team is ready to begin a template:
      - Move the ticket into the Development column
      - Add the team doing the work to Implementing team
      - Add the individual Assignee
      - Add a comment in the ticket informing the appropriate folks that this ticket is moving forward
    - When the template is ready for QA:
      - Move the ticket into the QA column
      - If the CMS team is migrating the template - change the Assignee to Laura
      - If the Sitewide team is migrating the template…(should we specific anything here?)
      - IN PROGRESS: QA engineer will follow the documented QA process/template
    - If there are issues found during QA
      - Create a Defect ticket for each issue found
      - These tickets should be added to the Defect column
      - These tickets should be subissues of the migration template ticket
      - The Parent template ticket should be moved back to Development once QA is complete and all issues are documented
      - Change the Assignee back to the original engineer
    - Development/QA cycle
      - The engineer will address all issues found
      - Once issues are resolved, the ticket should be moved back to QA and assigned to the QA engineer
      - This cycle of development and testing continues until all issues are resolved and no new issues are found
    - Code review?
      - Need to determine if this step/column is needed and if so when in the process
    - When the template is ready for Rollout
      - Move the ticket to the Rollout column
      - Make sure it has the correct Assignee
      - Add a comment in the ticket informing the appropriate folks that this ticket is moving
    - Moved to production
      - Is this column still needed?
      - Is it a step in the Rollout phase? Meaning, the template has launched but we are monitoring for a sprint to make sure no issues are reported, then we need to remove content-build template, switch codeowners, etc. Then it can close?
    - What does it mean that a ticket is ready to Close?
      - All ACs for each phase are complete and checked off in within the ticket
      - The template has been successfully deployed to production without report of issues for 1 full sprint
      - The content-build templates have been successfully removed
      - The CODEOWNERS file has been updated to shift ownership to the Sitewide team
      - The template is now in the Maintenance & Enhancements phase
      - If there are any outstanding issues, they were agreed on as post-launch, Sitewide issues and been moved to their Backlog/board

- An understanding across teams of which Github projects and Slack channels will be used moving forward for the template migration work.
  - Slack:
    - Update name of `#accelerated-publishing` channel to `#next-build`
    - This channel should continue to be used for all work around next-build and template migration work
  - GitHub:
    - Continue using the [Next-build project board](https://github.com/orgs/department-of-veterans-affairs/projects/1464)
      - This has been reorganized to alone with the new and agreed upon process within the RACI
    - **IN PROGRESS:** Tickets in [the existing AP board](https://github.com/orgs/department-of-veterans-affairs/projects/955) should be folded into the Next-build board or the CMS board, wherever is most appropriate

**Comment by laflannery** [comment]
Will use 2/19 Sitewide/CMS sync to discuss with team
